src/script/squirrel.cpp

```diff
@@ -412,7 +412,7 @@ bool Squirrel::CallStringMethod(HSQOBJECT instance, std::string_view method_name
 	HSQOBJECT ret;
 	if (!this->CallMethod(instance, method_name, &ret, suspend)) return false;
 	if (ret._type != OT_STRING) return false;
-	*res = StrMakeValid(ObjectToString(&ret));
+	*res = StrMakeValid(*ObjectToString(&ret));
```

Comment:
Does this need to check `has_value`?

Reply:
No. `ObjectToString` calls `sq_objtostring` which on type `OT_STRING` returns a `std::string_view` and anything other than `OT_STRING` it returns `std::nullopt`. Though... I can make the code itself clearer.